modules/cnf-understanding-templates.adoc

```diff
@@ -0,0 +1,105 @@
+// Module included in the following assemblies:
+// Epic CNF-9657 (TELCODOCS-1540) (4.16)
+// * edge_computing/cnf-understanding-siteconfig-operator.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="cnf-understanding-templates_{context}"]
+= Understanding templates
+
+The {sco} supports the inclusion of Golang text templates in the object definitions, which allows you to dynamically generate installation manifests for target clusters.
+This reduces the need to create separate manifests for many clusters with similar configurations but with different values.
+
+Additionally, the {sco} supports all link:https://masterminds.github.io/sprig/[sprig library functions] and provides two new functions that you can use while creating your custom manifests:
+
+`toYaml`:: The `toYaml` function encodes an item into a YAML string. If the item cannot be converted to YAML the function returns an empty string.
+
+`anyFieldDefined`:: The `anyFieldDefined` function checks if a specific field is defined. If defined, the {sco} renders the field as instructed.
+
+.anyFieldDefined example from InfraEnv
+[source,yaml]
+----
+[...]
+{{ if (anyFieldDefined .Site.Proxy) }}
+    proxy:
+{{ .Site.Proxy | toYaml | indent 4 }}
+{{ end }}
+[...]
+----
+
+There are two types of templates:
```

Comment:
Would be great if we can flow into this section... sensing a bit of disconnect in the readability.

Reply:
I have moved this up a bit and rephrased the intro to the new functions to connect them